### DISCUSSION TITLE
Aggiornamento fallito di Zigbee2MQTT con SONOFF Zigbee 3.0 USB Dongle Plus

### DISCUSSION THREAD

**Discussion by Newreader** [indomus/forum#1626] [GENERALE - Discussioni generiche]
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:\Users\rbvn0\AppData\Local\Temp\msohtmlclip1\01\clip_filelist.xml">

<link rel=themeData
href="file:///C:\Users\rbvn0\AppData\Local\Temp\msohtmlclip1\01\clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:\Users\rbvn0\AppData\Local\Temp\msohtmlclip1\01\clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=IT style='tab-interval:35.4pt'>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Salve,<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>qualche
tempo fa ho sostituito come coordinator la pennetta CC2531 con il più performante
SONOFF Zigbee 3.0 USB Dongle Plus. Ho apportato le opportune modifiche al file
configuration .yaml che nella mia installazione si trova sotto la directory /opt/zigbee2mqtt/data
<span style='mso-spacerun:yes'> </span>(in particolare ho sostituito a ttyACM0 il
valore <span style='mso-spacerun:yes'> </span>ttyUSB0). Tutto ha funzionato
perfettamente.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Adesso mi
trovo ad aggiornare la versione di Zigbee2MQTT alla 1.25.0 e purtroppo ho avuto
dei problemi.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Dopo aver
lanciato i canonici comandi:<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-ansi-language:EN-US'>sudo docker container stop
zigbee2mqtt<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>sudo docker
rm -f zigbee2mqtt<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>sudo docker
rmi -f koenkk/zigbee2mqtt:latest<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Successivamente
eseguo l’ulteriore comando per Istanziare ZigBee2MQTT su Docker<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-ansi-language:EN-US'>sudo docker run --init -d
--name=&quot;zigbee2mqtt&quot; -e TZ=Europe/Rome -v /opt/zigbee2mqtt/data:/app/data
--device=/dev/ttyUSB0 --restart unless-stopped --network host
koenkk/zigbee2mqtt<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-ansi-language:EN-US'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>e lancio per
verifica:<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>sudo docker
container logs zigbee2mqtt –f<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>ottenendo
come risultato:<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-ansi-language:EN-US'>[WARN<span
style='mso-spacerun:yes'>  </span>tini (8)] Tini is not running as PID 1 and
isn't registered as a child subreaper.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Zombie
processes will not be re-parented to Tini, so zombie reaping won't work.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>To fix the problem,
use the -s option or set the environment variable TINI_SUBREAPER to register
Tini as a child subreaper, or run Tini as PID 1.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>#<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'># Fatal
error in , line 0<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>#
unreachable code<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>#<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>#FailureMessage
Object: 0x7ed3563c<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Using
'/app/data' as data directory<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>[WARN<span
style='mso-spacerun:yes'>  </span>tini (8)] Tini is not running as PID 1 and
isn't registered as a child subreaper.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Zombie
processes will not be re-parented to Tini, so zombie reaping won't work.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>To fix the
problem, use the -s option or set the environment variable TINI_SUBREAPER to
register Tini as a child subreaper, or run Tini as PID 1.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Cosa che mi
è accaduta per la prima volta da quando aggiorno Zigbee2MQTT.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>La presenza
di –init nel comando docker run ndica che un processo init deve essere utilizzato
come PID 1 quindi in contrasto con quanto riportato nei logs.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Per inciso
non ho mai installato Tini nel mio sistema che ha queste caratteristiche:<o:p></o:p></p>

Versione | core-2022.4.1
-- | --
Tipo di   installazione | Home   Assistant Container
Sviluppo | false
Supervisore | false
Docker | true
Utente | root
Ambiente   virtuale | false
Versione   Python | 3.9.9
Famiglia   del sistema operativo | Linux
Versione   del sistema operativo | 5.4.83-v7+
Architettura   della CPU | armv7l
Fuso   orario | Europe/Rome

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Installato su
un Raspberry Pi 3 Model B+<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Potete darmi
una mano a risolvere il problema.<o:p></o:p></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt'>Grazie<o:p></o:p></p>

</body>

</html>

**Comment by Newreader** [comment]
Ho risolto leggendo la documentazione di Zigbee2MQTT:

Note per gli utenti di Raspberry Pi
Se stai eseguendo Raspbian Buster (non Bullseye!) (scopri eseguendo grep "PRETTY_NAME" /etc/os-release) devi installare libseccomp2, questo può essere fatto eseguendo i seguenti comandi:

sudo apt-key adv --keyserver hkps://keyserver.ubuntu.com:443 --recv-keys 04EE7237B7D453EC 648ACFD622F3D138
echo "deb http://httpredir.debian.org/debian buster-backports main contrib non-free" | sudo tee -a "/etc/apt/sources.list.d/debian-backports.list"
sudo apt update
sudo apt install libseccomp2 -t buster-backports

In caso contrario, all'avvio del contenitore Zigbee2MQTT verrà visualizzato il seguente errore:
Fatal error in, line 0
unreachable code
FilureMessage Object: 0x7eace25c

**Comment by indomus** [comment]
nella nostra guida di installazione è indicato 👍
